test/test-parse-warning.cpp

```diff
@@ -169,6 +169,162 @@ TEST_F(test_error_equals_does_not_distribute_over_or, non_constant) {
     EXPECT_THAT(p.errors, IsEmpty());
   }
 }
+
+TEST_F(test_parse_warning, warn_on_pointless_string_compare) {
+  {
+    padded_string code(u8"toLowerCase() == 'banana'"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.errors, IsEmpty());
```

Comment:
Nit: I changed the style of tests recently. Use `test_parser` instead of creating a `parser`, `spy_visitor`, and `padded_string` manually.